export_weekly_non_relevant_contacts.py

```diff
@@ -0,0 +1,142 @@
+                             "Exported file is in a format suitable for direct upload to Rapid Pro")
+
+    args = parser.parse_args()
+
+    exclusion_list_file_path = args.exclusion_list_file_path
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    target_dataset_name = args.target_dataset_name
+    traced_data_paths = args.traced_data_paths
+    csv_output_file_path = args.csv_output_file_path
+
+    sys.setrecursionlimit(10000)
+
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+    Logger.set_project_name(pipeline_configuration.pipeline_name)
+    log.debug(f"Pipeline name is {pipeline_configuration.pipeline_name}")
+
+    log.info("Downloading Firestore UUID Table credentials...")
+    firestore_uuid_table_credentials = json.loads(google_cloud_utils.download_blob_to_string(
+        google_cloud_credentials_file_path,
+        pipeline_configuration.uuid_table.firebase_credentials_file_url
+    ))
+
+    phone_number_uuid_table = FirestoreUuidTable.init_from_credentials(
+        firestore_uuid_table_credentials,
+        pipeline_configuration.uuid_table.table_name,
+        pipeline_configuration.uuid_table.uuid_prefix
+    )
+    log.info("Initialised the Firestore UUID table")
+
+    uuids = set()
+    for path in traced_data_paths:
+        # Load the traced data
+        log.info(f"Loading previous traced data from file '{path}'...")
+        with open(path) as f:
+            data = TracedDataJsonIO.import_jsonl_to_traced_data_iterable(f)
+        log.info(f"Loaded {len(data)} traced data objects")
+
+        for td in data:
+            has_relevant_message = False
+            if td["consent_withdrawn"] == Codes.TRUE:
+                continue
+
+            # Check for contacts who have only sent messages labeled as showtime_question", "greeting", "opt_in", "NC"
+            # in the target episodes. We will send a tailored follow up sms with the target episode question
+            for plan in PipelineConfiguration.RQA_CODING_PLANS:
+                if plan.dataset_name == target_dataset_name:
+                    for cc in plan.coding_configurations:
+                        codes = analysis_utils.get_codes_from_td(td, AnalysisConfiguration(plan.dataset_name,
+                                                                                           plan.raw_field,
+                                                                                           cc.coded_field,
+                                                                                           cc.code_scheme))
+                        for code in codes:
+                            if code.code_type == CodeTypes.NORMAL:
+                                has_relevant_message = True
```

Comment:
CoreData has an implementation of relevance that might help here

Reply:
Updated thanks.